packages/provider/src/MultiChainProvider.ts

```diff
@@ -0,0 +1,137 @@
+import SafeEventEmitter from '@metamask/safe-event-emitter';
+import { nanoid } from 'nanoid';
+import {
+  assertIsConnectArguments,
+  assertIsMetaMaskNotification,
+  assertIsMultiChainRequest,
+  assertIsSession,
+  ChainId,
+  ConnectArguments,
+  NamespaceId,
+  RequestArguments,
+  RequestNamespace,
+  Session,
+} from '@metamask/snap-utils';
+import { assertIsJsonRpcSuccess, JsonRpcRequest } from '@metamask/utils';
+import type { SnapProvider } from '@metamask/snap-types';
+import { Provider } from './Provider';
+
+declare global {
+  // Declaration merging doesn't work with types, so we have to use an interface
+  // here.
+  // eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+  interface Window {
+    ethereum: SnapProvider;
```

Comment:
FYI - this is `MetaMaskInPageProvider`, you can import `@metamask/providers` as a `devDepedency` and just use it as a type